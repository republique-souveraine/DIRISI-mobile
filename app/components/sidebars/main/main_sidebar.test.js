// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {shallow} from 'enzyme';

import Preferences from 'mattermost-redux/constants/preferences';

import ChannelSidebar from './main_sidebar';

jest.mock('react-intl');

describe('ChannelSidebar', () => {
    const baseProps = {
        actions: {
            getTeams: jest.fn(),
            logChannelSwitch: jest.fn(),
            makeDirectChannel: jest.fn(),
            setChannelDisplayName: jest.fn(),
            setChannelLoading: jest.fn(),
        },
        blurPostTextBox: jest.fn(),
        currentTeamId: 'current-team-id',
        currentUserId: 'current-user-id',
        deviceWidth: 10,
        isLandscape: false,
        teamsCount: 2,
        theme: Preferences.THEMES.default,
    };

    test('should match, full snapshot', () => {
        const wrapper = shallow(
            <ChannelSidebar {...baseProps}/>
        );

        expect(wrapper.getElement()).toMatchSnapshot();
    });
});
