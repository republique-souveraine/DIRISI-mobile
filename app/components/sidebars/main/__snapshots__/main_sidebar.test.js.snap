// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChannelSidebar should match, full snapshot 1`] = `
<DrawerLayout
  drawerPosition="left"
  drawerWidth={-30}
  isTablet={false}
  onDrawerClose={[Function]}
  onDrawerOpen={[Function]}
  renderNavigationView={[Function]}
  useNativeAnimations={true}
/>
`;
